src/support/azext_support/custom.py

```diff
@@ -272,3 +280,102 @@ def pre_operations(self):
         super().pre_operations()
         args = self.ctx.args
         _check_name_availability_no_subscription(self.cli_ctx, str(args.communication_name), "Microsoft.Support/communications")
+
+class FileWorkspaceCreateNoSubscription(_CreateNoSubscriptionFileWorkspace):
+    def pre_operations(self):
+        from azext_support._validators import _check_name_availability_no_subscription
+        super().pre_operations()
+        args = self.ctx.args
+        file_workspace_name = str(args.file_workspace_name)
+        _check_name_availability_no_subscription(self.cli_ctx, file_workspace_name, "Microsoft.Support/workspace")
+    
+class FileWorkspaceCreateSubscription(_CreateFileWorkspace):
+    def pre_operations(self):
+        from azext_support._validators import _check_name_availability_subscription
+        super().pre_operations()
+        args = self.ctx.args
+        file_workspace_name = str(args.file_workspace_name)
+        _check_name_availability_subscription(self.cli_ctx, file_workspace_name, "Microsoft.Support/workspace")
+    
+def encode_string_content(chunk_content):
+    return str(base64.b64encode(chunk_content).decode('utf-8'))
+
+def get_file_content(file_path):
+    with open(file_path, 'rb') as file:
+        content_bytes = file.read()
+    return content_bytes
+
+def get_file_name_info(file_path):
+    directory, full_file_name = os.path.split(file_path)
+    file_name_without_extension, file_extension = os.path.splitext(full_file_name)
+    return full_file_name, file_name_without_extension, file_extension
+
+def upload_files_no_subscription(cmd, file_path, file_workspace_name):
+    from .aaz.latest.support.no_subscription.file import Create
+    from .aaz.latest.support.no_subscription.file import Upload
+
+    ##costants for file upload
+    max_chunk_size= 2621440
+
+    validate_file_path(file_path)
+
+    full_file_name, file_extension, file_name_without_extension = get_file_name_info(file_path)
+    content = get_file_content(file_path)
+
+    validate_file_extension(file_extension)
+    validate_file_name(file_name_without_extension)
+    
+    file_size = len(content)
```

Comment:
Group similar functions together and assign them as regions for distinction. For eg: once you get content for file you call validate filesize, but I see validate extension and filename is called in between, logically group them together and optimize this flow and break them down into regions for ease of understanding